compose/project.py

```diff
@@ -145,6 +171,21 @@ def get_services_without_duplicate(self, service_names=None, include_deps=False)
             service.remove_duplicate_containers()
         return services
 
+    def get_networks(self, service_dict, network_definitions):
+        networks = []
+        for name in service_dict.pop('networks', ['default']):
+            if name in ['bridge', 'host']:
+                networks.append(name)
+            else:
+                matches = [n for n in network_definitions if n.name == name]
+                if matches:
+                    networks.append(matches[0].full_name)
+                else:
+                    raise ConfigurationError(
+                        'Service "{}" uses an undefined network "{}"'
+                        .format(service_dict['name'], name))
+        return networks
```

Comment:
Not a blocker, but this doesn't use `self` so it doesn't need to be a method on `Project` at all. It can be just a function at the module level.

I think that's good for consistency as well, since it's called from a `@classmethod` for the `Project` is fully initialized.

Maybe it could actually be a function in the `network.py` module ?